robottelo/ssh.py

```diff
@@ -73,18 +76,28 @@ def _get_connection(
         hostname = settings.server.hostname
     if username is None:
         username = settings.server.ssh_username
-    if key_filename is None:
+    if key_filename is None and password is None:
         key_filename = settings.server.ssh_key
+        # If we don't have key even in settings, read password instead:
+        if key_filename is None:
+            password = settings.server.ssh_password
```

Comment:
Got it! @oshtaier, so the ``settings.server.ssh_password`` is lazy?

It can be 

```python
if key_filename is None:
    key_filename = settings.server.ssh_key
# DROP the below
#if password is None:
#    password = settings.server.ssh_password

# And read then lazily in one-liner here
client.connect(
    hostname=hostname,
    username=username,
    key_filename=key_filename,
    password=password or settings.server.ssh_password if key_filename is None else None,
    timeout=timeout
)
```

**if** is avalueted first and **or** is short-circuit so if key exists the **read** operation will not happen.

But I don't know yet what are the policy about one-liners